racket/collects/xml/private/xexpr-core.rkt

```diff
@@ -75,6 +77,11 @@
       (true-k)))
 
 
+(define (non-empty-symbol? x)
+  (and
+   (symbol? x)
+   (non-empty-string? (symbol->string x))))
```

Comment:
Nit: I would replace `symbol->string` with `symbol->immutable-string` to avoid an unnecessary allocation and copy. 

The `symbol->string` implementation is [about 10x slower](https://gist.github.com/Bogdanp/9ada3e8dec7e274d47c38e971b5bb934) compared to an `eq`-based implementation (which I assume you didn't go with due to `(string->uninterned-symbol "")`), and a `symbol->immutable-string` version is only 5x slower.

Also, it's probably worth adding a couple of tests for this change.

Reply:
Maybe a short-circuit with and `eq?` then? Something like this?
```
#lang racket/base
(require racket/symbol)

(define (non-empty-symbol? x)
  (cond
    [(symbol? x)
     (cond
       [(symbol-interned? x)
        (not (eq? x '||))]
       [else
        (not (= 0 (string-length (symbol->immutable-string x))))])]
    [else #f]))

(require rackunit)
(module+ test
  (check-true (non-empty-symbol? 'x))
  (check-false (non-empty-symbol? '||))
  (check-false (non-empty-symbol? 0))
  (check-false (non-empty-symbol? (string->uninterned-symbol "")))
  (check-true (non-empty-symbol? (string->uninterned-symbol "x"))))
```

Reply:
I have gone with @rfindler's suggestion. I'm unsure if I have put the unit tests in the right spot though. If someone has a better place than a `(module+ test ...)` in the same file, I'm happy to go with that.